src/Microsoft.AspNetCore.Routing/RouteBase.cs

```diff
@@ -46,12 +46,19 @@ public abstract class RouteBase : IRouter, INamedRouter
             ConstraintResolver = constraintResolver;
             DataTokens = dataTokens ?? new RouteValueDictionary();
 
-            // Data we parse from the template will be used to fill in the rest of the constraints or
-            // defaults. The parser will throw for invalid routes.
-            ParsedTemplate = TemplateParser.Parse(template);
+            try
+            {
+                // Data we parse from the template will be used to fill in the rest of the constraints or
+                // defaults. The parser will throw for invalid routes.
+                ParsedTemplate = TemplateParser.Parse(template);
 
-            Constraints = GetConstraints(constraintResolver, ParsedTemplate, constraints);
-            Defaults = GetDefaults(ParsedTemplate, defaults);
+                Constraints = GetConstraints(constraintResolver, ParsedTemplate, constraints);
+                Defaults = GetDefaults(ParsedTemplate, defaults);
+            }
+            catch (Exception exception)
```

Comment:
Yeah this was intentional because any RouteCreationException thrown from within constraints or other methods here would not give contextual information like route name and template and to accommodate that we might need to pass on those details around.